src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -710,9 +712,29 @@ public static SocketError Poll(SafeCloseSocket handle, int microseconds, SelectM
 
         public static SocketError Select(IList checkRead, IList checkWrite, IList checkError, int microseconds)
         {
-            IntPtr[] readfileDescriptorSet = Socket.SocketListToFileDescriptorSet(checkRead);
-            IntPtr[] writefileDescriptorSet = Socket.SocketListToFileDescriptorSet(checkWrite);
-            IntPtr[] errfileDescriptorSet = Socket.SocketListToFileDescriptorSet(checkError);
+            bool ShouldStackAlloc(IList list, out Span<IntPtr> span)
+            {
+                const int StackThreshold = 80; // arbitrary limit to avoid too much space on stack
+                if (list == null || list.Count == 0)
+                {
+                    span = default;
+                    return false;
+                }
+                if (list.Count >= StackThreshold)
+                {
+                    span = new IntPtr[list.Count + 1];
```

Comment:
`list.Count` is an interface invocation... worst case here, we're calling it three times when we really only need to call it once.  We should store the result in a local.

Reply:
done